src/internal.rs

```diff
@@ -373,19 +373,61 @@ pub type cdef_fn = Option<
 pub type cdef_dir_fn =
     Option<unsafe extern "C" fn(*const libc::c_void, ptrdiff_t, *mut libc::c_uint) -> libc::c_int>;
 
-pub type const_left_pixel_row = *const libc::c_void;
-pub type looprestorationfilter_fn = Option<
-    unsafe extern "C" fn(
-        *mut libc::c_void,
-        ptrdiff_t,
-        const_left_pixel_row,
-        *const libc::c_void,
-        libc::c_int,
-        libc::c_int,
-        *const LooprestorationParams,
-        LrEdgeFlags,
-    ) -> (),
->;
+pub type pixel_8bpc = uint8_t;
+pub type coef_8bpc = int16_t;
+
+pub type pixel_16bpc = uint16_t;
+pub type coef_16bpc = int32_t;
+
+pub type const_left_pixel_row<Pixel> = *const [Pixel; 4];
+
+pub type const_left_pixel_row_8bpc = const_left_pixel_row<pixel_8bpc>;
+pub type looprestorationfilter_fn_8bpc = unsafe extern "C" fn(
+    *mut pixel_8bpc,
+    ptrdiff_t,
+    const_left_pixel_row_8bpc,
+    *const pixel_8bpc,
+    libc::c_int,
+    libc::c_int,
+    *const LooprestorationParams,
+    LrEdgeFlags,
+) -> ();
+
+pub type const_left_pixel_row_16bpc = const_left_pixel_row<pixel_16bpc>;
+pub type looprestorationfilter_fn_16bpc = unsafe extern "C" fn(
+    *mut pixel_16bpc,
+    ptrdiff_t,
+    const_left_pixel_row_16bpc,
+    *const pixel_16bpc,
+    libc::c_int,
+    libc::c_int,
+    *const LooprestorationParams,
+    LrEdgeFlags,
+    libc::c_int,
+) -> ();
+
+#[derive(Clone, Copy)]
+#[repr(C)]
```

Comment:
Once these are data-less `enum`s, I think we can make it `#[repr(u8)]` instead of `#[repr(C)]`.  `#[repr(u8)]` is also stable, just smaller.